### DISCUSSION TITLE
Correct way to migrate dev vs production?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#17938] [Q&A]
Lately I have been having trouble understanding the correct way to save schema changes. I use 3 different computers, and whenever I update the schema in one computer, it prompts me the following message:
- The migrations recorded in the database diverge from the local migrations directory.
- Do you want to continue? All data will be lost

Right now I am in developing mode, but I keep resetting the complete database anytime I use a different computer to push changes. 

The steps I normally take:
I run npx prisma migrate dev and it prompts me to erase everything because it does not matches the new pulled branch I am working on. 
I erase it and start all over again with new changes
Do that thing all over again with other computers.

Running prisma status displays following message, which is basically what happens every time:
The migration have not yet been applied:
20230207103839_first

The migration from the database are not found locally in prisma/migrations:
20230208141707_initial

**Comment by ludralph** [comment]
Hi @gmcamposano 👋 

It's generally a good practice to create and apply migrations locally on each computer as you change your schema. Then, you can commit and push these migrations to your version control system (e.g., Git). You use the `migrate dev` command to create and apply migrations in a development environment.  On the other computer, you can pull these changes and apply them to their local databases. This can help ensure that all computers are working from the same schema and that there are no conflicts between different database versions.

Suppose you encounter the "The migrations recorded in the database diverge from the local migrations directory" message. In that case, it typically means that there are migrations that have been applied to your database that are not in your local migrations directory. This can happen if you've reset your database or deleted applied migrations without updating your local directory. In this case, you can continue with the "All data will be lost" option, but this will erase all data in your database and may cause other issues.

If you're frequently resetting your database to apply schema changes, you may consider using seeds to populate your database with sample data. This can help you test your application with more realistic data and reduce the need to reset your database constantly.

You can look at this [section](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate) of the documentation for a typical development workflow with Prisma migrate.

**Reply by gmcamposano** [reply]
Hello! Thank you for your response. 

I will take that into consideration for now on. As for now, how can I fix, without having to reset the DB? Is there a way to include the missing migration and from there on `prisma migrate dev` or something similar to it?

So basically, whenever I get this message, which i hope not to get ever again, how do i solve it without having to reset the db?
The migrations recorded in the database diverge from the local migrations directory.
Do you want to continue? All data will be lost

**Reply by ludralph** [reply]
You might want to consider [baselining](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining#baselining-a-database) since you do not want to reset the database.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
